modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/portlet/action/PDFPreviewConfigurationDisplayContext.java

```diff
@@ -50,8 +54,32 @@ public String getEditPDFPreviewConfigurationURL() {
 		).buildString();
 	}
 
+	public long getMaxNumberPages() {
+		if (_scope.equals(
+				ExtendedObjectClassDefinition.Scope.COMPANY.getValue())) {
+
+			return _pdfPreviewManagedServiceFactory.getCompanyMaxNumberPages(
+				_scopePK);
+		}
+		else if (_scope.equals(
+					ExtendedObjectClassDefinition.Scope.GROUP.getValue())) {
+
+			return _pdfPreviewManagedServiceFactory.getGroupMaxNumberPages(
+				_scopePK);
+		}
+		else if (_scope.equals(
+					ExtendedObjectClassDefinition.Scope.SYSTEM.getValue())) {
+
+			return _pdfPreviewManagedServiceFactory.getSystemMaxNumberPages();
+		}
+
+		throw new IllegalArgumentException("Unsupported scope: " + _scope);
```

Comment:
Can you move this code into the managed service factory? Pass in the scope and scopePK and do the checks internally. That way, callers don't have to worry about how many levels of configuration there are.